File: 66-plus-one/plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
         reverse(digits.begin(), digits.end());
       int size=digits.size();

       bool all9=true;
       for(int i=0;i<size;i++){if(digits[i]!=9){all9=false;}}


       if(all9){digits.push_back(0);digits[size]=1;
       for(int i=0;i<size;i++){digits[i]=0;}}

      else{digits[0]=digits[0]+1;
      for(int i=0;i<size;i++){
        if(digits[i]==10){digits[i+1]=digits[i+1]+1;
        digits[i]=0;}
      } }
      reverse(digits.begin(),digits.end());
        return digits;
    }
};

File: 66-plus-one/plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        n = len(digits)
        if all([digits[i]==9 for i in range(n)]):
            digits = [0]*(n + 1)
            digits[0] = 1
            return digits
        digits[-1] += 1
        for i in range(n-1 , -1 , -1):
            print(digits)
            if digits[i] > 9:
                digits[i] = digits[i]%10
                digits[i-1] += 1
            else:
                break
        return digits
        

